src/Mesh/babylon.subMesh.ts

```diff
@@ -353,13 +355,15 @@ module BABYLON {
         }
 
         /** @hidden */
-        private _intersectLines(ray: Ray, positions: Vector3[], indices: IndicesArray, intersectionThreshold: number, fastCheck?: boolean): Nullable<IntersectionInfo> {
+        private _intersectLines(ray: Ray, positions: FloatArray, indices: IndicesArray, intersectionThreshold: number, fastCheck?: boolean): Nullable<IntersectionInfo> {
             var intersectInfo: Nullable<IntersectionInfo> = null;
+            const p0 = SubMesh.TmpVector3[0]
+            const p1 = SubMesh.TmpVector3[1]
 
             // Line test
             for (var index = this.indexStart; index < this.indexStart + this.indexCount; index += 2) {
-                var p0 = positions[indices[index]];
-                var p1 = positions[indices[index + 1]];
+                Vector3.FromArrayToRef(positions, 3*indices[index], p0)

```

Comment:
My only concern is the cost of this change as we new to copy 6 values per face (and 9 for triangles later).

Can you do a benchmark to make sure we are not slower than before?

Reply:
Well we'll probably be a bit slower than before as we do more operations. I'll try to measure how much slower so that we can weight the trade-off between CPU "loss" Vs Memory "gain". My gut feeling is that it should not be that slower.

@jbousquie, it seems you've got already a lot of scenes/test cases, is there one that might be used for benchmarking the picking ? @deltakosh Do you have something already ? If no I'll build one

Reply:
I don't out of the box unfortunately

Reply:
up?

Reply:
down 😞 
Pretty busy these days, did not find time so far to test.

Reply:
Ok  no worry  
Ping me when you have a chance to do it